experimental/QuadFormAndIsom/src/enumeration.jl

```diff
@@ -0,0 +1,952 @@
+  else
+    return Oscar._anti_isometry_bilinear(rA, rB)[1]
+  end
+end
+
+function is_admissible_triple(A::T, B::T, C::T, p::Integer) where T <: Union{ZZLat, ZZLatWithIsom}
+  L = ZZGenus[genus(D) for D = (A, B, C)]
+  return is_admissible_triple(L[1], L[2], L[3], p)
+end
+
+@doc raw"""
+    admissible_triples(C::ZZGenus, p::Integer) -> Vector{Tuple{ZZGenus, ZZGenus}}
+
+Given a $\mathbb Z$-genus `C` and a prime number `p`, return all tuples of
+$\mathbb Z$-genera `(A, B)` such that `(A, B, C)` is `p`-admissible and
+`B` is of rank divisible by $p-1$.
+
+# Examples
+```jldoctest
+julia> L = root_lattice(:A,5);
+
+julia> g = genus(L)
+Genus symbol for integer lattices
+Signatures: (5, 0, 0)
+Local symbols:
+  Local genus symbol at 2: 1^-4 2^1_7
+  Local genus symbol at 3: 1^-4 3^1
+
+julia> admissible_triples(g, 5)
+2-element Vector{Tuple{ZZGenus, ZZGenus}}:
+ (Genus symbol: II_(5, 0) 2^-1_3 3^1, Genus symbol: II_(0, 0))
+ (Genus symbol: II_(1, 0) 2^1_7 3^1 5^1, Genus symbol: II_(4, 0) 5^1)
+
+julia> admissible_triples(g, 2)
+8-element Vector{Tuple{ZZGenus, ZZGenus}}:
+ (Genus symbol: II_(5, 0) 2^-1_3 3^1, Genus symbol: II_(0, 0))
+ (Genus symbol: II_(4, 0) 2^2_6 3^1, Genus symbol: II_(1, 0) 2^1_1)
+ (Genus symbol: II_(3, 0) 2^3_3, Genus symbol: II_(2, 0) 2^-2 3^1)
+ (Genus symbol: II_(3, 0) 2^-3_1 3^1, Genus symbol: II_(2, 0) 2^2_2)
+ (Genus symbol: II_(2, 0) 2^2_2, Genus symbol: II_(3, 0) 2^-3_1 3^1)
+ (Genus symbol: II_(2, 0) 2^-2 3^1, Genus symbol: II_(3, 0) 2^3_3)
+ (Genus symbol: II_(1, 0) 2^1_1, Genus symbol: II_(4, 0) 2^2_6 3^1)
+ (Genus symbol: II_(0, 0), Genus symbol: II_(5, 0) 2^-1_3 3^1)
+```
+"""
+function admissible_triples(G::ZZGenus, p::Integer; pA::Int = -1, pB::Int = -1)
+  @req is_prime(p) "p must be a prime number"
+  @req is_integral(G) "G must be a genus of integral lattices"
+  n = rank(G)
+  pG, nG = signature_pair(G)
+  if pA >= 0
+    @req pA <= pG "Wrong restrictions"
+    if pB >= 0
+      @req pA + pB == pG "Wrong restrictions"
+    else
+      pB = pG - pA
+    end
+  elseif pB >= 0
+    @req pB <= pG "Wrong restrictions"
+    pA = pG - pB
+  end
+  d = numerator(det(G))
+  even = iseven(G)
+  L = Tuple{ZZGenus, ZZGenus}[]
+  for ep in 0:div(n, p-1)
+    rp = (p-1)*ep
+    if pB >= 0
+      rp >= pB || continue
+    end
+    r1 = n - rp
+    if pA >= 0
+      r1 >= pA || continue
+    end
+    m = min(ep, r1) 
+    D = _find_D(d, m, p)
+    for (d1, dp) in D
+      L1 = _find_L(pG, nG, r1, d1, numerator(scale(G)), numerator(level(G)), p, even, pos = pA)
+      Lp = _find_L(pG, nG, rp, dp, numerator(scale(G)), numerator(level(G)), p, even, pos = pB)
+      for (A, B) in [(A, B) for A in L1 for B in Lp]
+        if is_admissible_triple(A, B, G, p)
+          push!(L, (A, B))
+        end
+      end
+    end
+  end
+  return L
+end
+
+admissible_triples(L::T, p::Integer; pA::Int = -1, pB::Int = -1) where T <: Union{ZZLat, ZZLatWithIsom} = admissible_triples(genus(L), p, pA = pA, pB = pB)
```

Comment:
```suggestion
admissible_triples(L::T, p::Integer; pA::Int = -1, pB::Int = -1) where T <: Union{ZZLat, ZZLatWithIsom} = admissible_triples(genus(L), p; pA, pB)
```
If the variable has the same name as the kwarg, the duplication can be avoided.